pkg/controller/operators/catalog/deprecated.go

```diff
@@ -0,0 +1,37 @@
+package catalog
```

Comment:
Since we have IP retries now — deprecation is a terminal error. Should we categorize the error and fail the plan immediately if this comes up? Since we do not have that, the test sits around for a minute until it fails. Not a huge deal, but just thinking about it. 
Also @timflannagan  made the point around alerting — do we alert when an IP fails? should there be an alert in this PR as well? 
Lastly, in the PR now OLM still can watch for v1beta1 CRDs, Operator objects can consist of v1beta1 CRDs, should we strip all this code as well?

Reply:
Ah, I had glossed over this comment when I was reviewing and made a similar comment in https://github.com/operator-framework/operator-lifecycle-manager/pull/2251/files#r669890440. tl;dr I think we shouldn't be failing the installplan resource here, but it would be difficult to deal with skew downstream, so it's an understandable approach and I'm open to hearing other opinions.

Is there a situation where we wouldn't want to immediately fail an InstallPlan? Nothing immediate comes to my mind, but I just wanted to double-check whether that assumption holds up.

Reply:
> Is there a situation where we wouldn't want to immediately fail an InstallPlan?

Yes, there are sometimes transient errors (errors connecting to the apiserver) or object has been modified errors that pop up during installplan execution. We added the retries to try to overcome these types of errors. Some errors are terminal though, and don't have to be retried - I think the idea was to build an explicit list of these at some point. 

This PR isn't really that big, like if we wanted to make this downstream only effectively it could not be such a huge burden. But it does increase the complexity going forward somewhat -- one version of OLM supports a different set of APIs than the other.